File: Module4/Case/FuramaNew/src/main/java/thanh/code/models/Division.java
package thanh.code.models;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import java.util.Set;

@Entity
public class Division {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int divisionId;
    @NotBlank
    private String divisionName;

    @OneToMany(mappedBy = "division", cascade = CascadeType.ALL)
    private Set<Employee> employee;

    public Set<Employee> getEmployee() {
        return employee;
    }

    public void setEmployee(Set<Employee> employee) {
        this.employee = employee;
    }

    public int getDivisionId() {
        return divisionId;
    }

    public void setDivisionId(int divisionId) {
        this.divisionId = divisionId;
    }

    public String getDivisionName() {
        return divisionName;
    }

    public void setDivisionName(String divisionName) {
        this.divisionName = divisionName;
    }

    public Division() {
    }
}

File: Module5/Case/Furama/src/app/models/division.ts
export class Division implements IDivision{
  divisionId: number;
  divisionName: string;

  constructor(divisionId: number, divisionName: string) {
    this.divisionId = divisionId;
    this.divisionName = divisionName;
  }
}

export interface IDivision {
  divisionId: number;
  divisionName: string;
}
